torch/csrc/jit/backends/backend_debug_handler.h

```diff
@@ -117,16 +119,72 @@ namespace jit {
  *  [M.forward, source range] -> [aten::mul's source range] We need to track
  *  mul's source range and inlined CS both.
  */
+using DebugHandleType = int64_t;
 class TORCH_API BackendDebugHandleManager {
  public:
   BackendDebugHandleManager() = default;
   int64_t getNextDebugHandleForInlinedCallStackPtr(const Node* node);
-  std::unordered_map<int64_t, DebugInfoPair> getCallStackPtrMap();
+  std::unordered_map<DebugHandleType, DebugInfoPair> getCallStackPtrMap();
+
+ private:
+  static std::atomic<DebugHandleType> unique_debug_handle_;
+  std::unordered_map<DebugHandleType, DebugInfoPair>
+      handles_to_inlined_callstack_ptrs_;
+};
+
+// Copied from torch/csrc/jit/api/object.h
+// to avoid including that file.
+using ObjectPtr = c10::intrusive_ptr<c10::ivalue::Object>;
+using DelegateDebugInfoMapType =
+    std::unordered_map<DebugHandleType, DebugInfoPair>;
+
+/*
+ * This class is used to generate debug info map (module's inline callstack ptr
+ * map). It is initialized with module's object ptr and during ctor it will
+ * instantiate debug_handle_manager and initialize thread local pointer to it.
+ * backend's preprocess will call generate_debug_handles, which uses
+ * debug_handle_manager to generate debug handles. When lowering process
+ * finishes, calling stopRecording will extract debug info map from
+ * debug_handle_manager and save it in an static global map to be
+ * used/serialized later.
+ */
+class BackendModuleDebugInfoRecorder {
+ public:
+  BackendModuleDebugInfoRecorder(ObjectPtr module_ptr);
```

Comment:
So actual module information is always inside Object. Module is just a wrapper around it. I did not chose Module because it would have required me to introduce hashing for Module which would effectively use Object anyway.
This would be a moot point though, now that we will move DebugInfoMap inside LoweredModule. So we dont need to provide ObjectPtr here at all.